jaeger-context/src/main/java/com/uber/jaeger/context/TracingUtils.java

```diff
@@ -14,18 +14,31 @@
 
 package com.uber.jaeger.context;
 
+import io.opentracing.util.GlobalTracer;
+
 import java.util.concurrent.ExecutorService;
 
 public class TracingUtils {
-  private static final TraceContext traceContext = new ThreadLocalTraceContext();
+  private static TraceContext traceContext;
 
   public static TraceContext getTraceContext() {
+    initializeTraceContext();
     return traceContext;
   }
 
   public static ExecutorService tracedExecutor(ExecutorService wrappedExecutorService) {
+    initializeTraceContext();
     return new TracedExecutorService(wrappedExecutorService, traceContext);
   }
 
-  private TracingUtils(){}
+  private static synchronized void initializeTraceContext() {
```

Comment:
in my suggestion there was no additional `synchronized` on the hot path

Reply:
I apologize, I didn't understand your previous suggestion, could you elaborate?

As such, `getTraceContext` and `tracedExecutor` are only used when initializing filters and executors. I expect these methods to be called a handful of times during application startup.

Reply:
instead of `ActiveSpanSourceTraceContext(Tracer)` declare `ActiveSpanSourceTraceContext(TracerProvider tracerProvider)` and then in the implementation every time you need a tracer you retrieve it dynamically via`tracerProvider.get()`, which will delegate to `GlobalTracer.get()`. This way you leave a lot more opportunities to the application to init the global tracer before it stumbles on the need to push/pop spans into TraceContext